hypothesis-python/src/hypothesis/internal/conjecture/shrinking/ordering.py

```diff
@@ -53,40 +53,18 @@ def run_step(self):
 
     def reinsert(self):
         for i in range(len(self.current)):
-            # This is essentially insertion sort, but unlike normal insertion
-            # sort because of our use of find_integer we only perform
-            # O(n(log(n))) calls. Because of the rotations we're still O(n^2)
-            # performance in terms of number of list operations, but we don't
-            # care about those so much.
-            original = self.current
-
             insertion_points = [
                 j
                 for j in range(i)
                 if self.key(self.current[j]) > self.key(self.current[i])
             ]
 
-            def push_back_to(t):
-                if t >= len(insertion_points):
-                    return True
-                j = insertion_points[t]
-                reinserted = list(original)
+            for j in insertion_points[:2]:
```

Comment:
It would be good to have some explanation of the tradeoff being made by this mysterious `:2`.

From what I can tell, it makes us try only the leftmost two plausible locations for reinsertion/swapping. I trust that this is helpful somehow, but it's not obvious why.

(My guess: `0` is the place the current item would go if the list were sorted. `1` is the next candidate, so that we have some flexibility to skip over essential parts that can't be reordered. After that, we throw up our hands and assume that a more exhaustive sorting attempt isn't going to be worth the effort.)